### PR TITLE
WIP: Math: Compute mimic joints' second-order partial derivatives, their second-order time derivatives, and link accelerations

### PR DESCRIPTION
**Pull request by tgn3000**
TBW.

### CODE REVIEW

File: src/libopenrave/kinbody.cpp [hunk 1]
```diff
@@ -3104,114 +3148,172 @@ void KinBody::ComputeInverseDynamics(std::vector<dReal>& doftorques, const std::
...
-            else if( pjoint->GetType() == JointSlider ) {
-                doftorques.at(pjoint->GetDOFIndex()) += pjoint->GetAxis(0).dot3(vcomforce)/(2*PI);
-            }
-            else {
-                throw OPENRAVE_EXCEPTION_FORMAT(_("joint 0x%x not supported"), pjoint->GetType(), ORE_Assert);
-            }
+        const Vector vcomtoanchor = pchildlink->GetGlobalCOM() - pjoint->GetAnchor();
+        const int jointdofindex = pjoint->GetDOFIndex();
+
+        if( jointdofindex >= 0 ) { // active
```
Comment (tgn3000) [inline src/libopenrave/kinbody.cpp:3240]: refactor this

File: src/libopenrave/kinbody.cpp [hunk 2]
```diff
@@ -3087,131 +3129,192 @@ void KinBody::ComputeInverseDynamics(std::vector<dReal>& doftorques, const std::
...
-        Vector vangularaccel = vLinkAccelerations.at(i).second;
-        Vector vangularvelocity = vLinkVelocities.at(i).second;
-        vLinkCOMLinearAccelerations[i] = vLinkAccelerations.at(i).first + vangularaccel.cross(vglobalcomfromlink) + vangularvelocity.cross(vangularvelocity.cross(vglobalcomfromlink));
-        TransformMatrix tm = _veclinks.at(i)->GetGlobalInertia();
-        vLinkCOMMomentOfInertia[i] = tm.rotate(vangularaccel) + vangularvelocity.cross(tm.rotate(vangularvelocity));
+        // collect
+        const LinkPtr& plink = _veclinks.at(i);
+        const Vector CoM = plink->GetGlobalCOM() - plink->_info._t.trans; // B-A
+        const Vector& angularvel = vLinkVelocities.at(i).second;
+        const Vector& linearvel = vLinkAccelerations.at(i).first;
```
Comment (yoshikikanemoto) [inline src/libopenrave/kinbody.cpp:3192]: linearaccel?

File: src/libopenrave/kinbody.cpp [hunk 3]
```diff
@@ -3658,35 +3776,96 @@ void KinBody::_ComputeLinkAccelerations(const std::vector<dReal>& vDOFVelocities
...
+
+            yields
+
+            alpha_child = alpha_parent
+
+            */
+
+            const Vector jointdir = tdelta.rotate(vlocalaxis);
+            // a_child = a_parent + alpha_parent x (p_child - p_parent) + w_parent x (w_parent x (p_child - p_parent))
+            childaccelerations.first = parentaccelerations.first + parentaccelerations.second.cross(xyzdelta) + parentvelocities.second.cross(parentvelocities.second.cross(xyzdelta));
```
Comment (yoshikikanemoto) [inline src/libopenrave/kinbody.cpp:3855]: looks correct for me too

**Comment by rdiankov** [comment]
@tgn3000 is this ready?

**Pull request closed without merging**